src/Serilog/Parameters/PropertyValueConverter.cs

```diff
@@ -121,11 +125,22 @@ LogEventPropertyValue CreatePropertyValue(object value, Destructuring destructur
                 return new ScalarValue(null);
 
             if (destructuring == Destructuring.Stringify)
-                return new ScalarValue(value.ToString());
+            {
+                return Stringify(value);
+            }
 
             var valueType = value.GetType();
             var limiter = new DepthLimiter(depth, _maximumDestructuringDepth, this);
 
+            if (destructuring == Destructuring.Destructure)
```

Comment:
I tried it and biggest drawback is that if we will use `IScalarConversionPolicy` as-is we are back to limiting strings in default mode Log.Information("The string is {S}", s); 
In my opinion this is good thing, strings would be limited in every case, no special cases.

Also the stringified mode is still going to be handled separately, but limiting logic was extracted to separate class, so no dupication there.

To sum up, I would go for this option. Are you sure you want to keep the requirement of `Log.Information("The string is {S}", s); will NOT be limited`?

https://github.com/krajek/serilog/commit/890f9be6cb54af0823fab68efc368a7ae27cce49
I have not changed limit to `int?`, it will be trivial.